File: Filesystem-Module/Filesystem-Module/UserControl_FileResource_File.vb
Imports Ontology_Module
Imports OntologyClasses.BaseClasses
Imports TextParser

Public Class UserControl_FileResource_File

    Private objLocalConfig As clsLocalConfig

    Private objOItem_FileResource As clsOntologyItem

    Private WithEvents objUserControl_Files As UserControl_OItemList
    Private WithEvents objUserControl_RegExTester As UserControl_RegExTester

    Public Sub New(LocalConfig As clsLocalConfig)

        ' Dieser Aufruf ist für den Designer erforderlich.
        InitializeComponent()

        ' Fügen Sie Initialisierungen nach dem InitializeComponent()-Aufruf hinzu.
        objLocalConfig = LocalConfig
        Initialize()
    End Sub

    Private Sub Initialize()
        objUserControl_Files = New UserControl_OItemList(objLocalConfig.Globals)
        objUserControl_Files.Dock = DockStyle.Fill
        SplitContainer1.Panel1.Controls.Add(objUserControl_Files)

        objUserControl_RegExTester = New UserControl_RegExTester(objLocalConfig.Globals)
        objUserControl_RegExTester.Dock = DockStyle.Fill
        SplitContainer1.Panel2.Controls.Add(objUserControl_RegExTester)
    End Sub

    Public Sub Initialize_File(OItem_FileResource As clsOntologyItem)
        objOItem_FileResource = OItem_FileResource
        If Not objOItem_FileResource Is Nothing Then
            Dim objOItem_File = New clsOntologyItem With {.GUID_Parent = objLocalConfig.OItem_Type_File.GUID, _
                                                          .Type = objLocalConfig.Globals.Type_Object}

            objUserControl_Files.initialize(Nothing, _
                                            objOItem_FileResource, _
                                            objLocalConfig.Globals.Direction_LeftRight, _
                                            objOItem_File, _
                                            objLocalConfig.OItem_RelationType_belonging_Resource)

        Else
            objUserControl_Files.clear_Relation()
        End If
    End Sub
End Class

File: FileResourceModule/FileResourceModule/UserControl_FileResource_File.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using OntologyClasses.BaseClasses;
using Ontology_Module;
using TextParser;

namespace FileResourceModule
{
    public partial class UserControl_FileResource_File : UserControl
    {
        private clsLocalConfig objLocalConfig;

        private UserControl_OItemList objUserControl_Files;
        private UserControl_RegExTester objUserControl_RegExTester;

        private clsOntologyItem objOItem_FileResource;

        public UserControl_FileResource_File(clsLocalConfig localConfig)
        {
            InitializeComponent();
            objLocalConfig = localConfig;
            Initialize();
        }

        private void Initialize()
        {
            objUserControl_Files = new UserControl_OItemList(objLocalConfig.Globals);
            objUserControl_Files.Dock = DockStyle.Fill;
            SplitContainer1.Panel1.Controls.Add(objUserControl_Files);

            objUserControl_RegExTester = new UserControl_RegExTester(objLocalConfig.Globals);
            objUserControl_RegExTester.Dock = DockStyle.Fill;
            SplitContainer1.Panel2.Controls.Add(objUserControl_RegExTester);

        }

        public void Initialize_File(clsOntologyItem OItem_FileResource)
        {
            objOItem_FileResource = OItem_FileResource;

            if (objOItem_FileResource != null)
            {
                var objOItem_File = new clsOntologyItem
                    {
                        GUID_Parent = objLocalConfig.OItem_class_file.GUID,
                        Type = objLocalConfig.Globals.Type_Object
                    };

                objUserControl_Files.initialize(null, objOItem_FileResource,
                                                objLocalConfig.Globals.Direction_LeftRight,
                                                objOItem_File,
                                                objLocalConfig.OItem_relationtype_belonging_resource);


            }
            else
            {
                objUserControl_Files.clear_Relation();
            }
        }
    }
}
